torch_geometric/nn/summary.py

```diff
@@ -0,0 +1,152 @@
+        x = torch.randn(100, 128)
+        edge_index = torch.randint(100, size=(2, 20))
+        print(summary(model, x, edge_index))
+
+    Args:
+        model (nn.Module): PyTorch model to summarize.
+        inputs (Sequence[Any]): Arguments for the model's
+            forward pass.
+        max_depth (int): Depth of nested layers to display.
+            Nested layers below this depth will not be displayed
+            in the summary. (default: :obj:`"3"`)
+        leaf_module (Union[nn.Module, Tuple[nn.Module], Optional[nn.Module]]):
+            Modules to be treated as leaf ones, whose submodules are excluded
+            from the the summarized information.
+            (default: :obj:`"MessagePassing"`)
+        **kwargs: Other arguments used in `model.forward` function.
+    """
+    def register_hook(info):
+        def hook(module, input, output):
+            info['input_shape'].append(get_shape(input))
+            info['output_shape'].append(get_shape(output))
+
+        return hook
+
+    hooks = {}
+    depth = 0
+    stack = [(get_name(model), model, depth)]
+
+    info_list = []
+    input_shape = defaultdict(list)
+    output_shape = defaultdict(list)
+    while stack:
+        var_name, module, depth = stack.pop()
+
+        module_id = id(module)
+        if module_id in hooks:
+            hooks[module_id].remove()
+
+        info = {}
+        info['name'] = var_name
+        info['input_shape'] = input_shape[module_id]
+        info['output_shape'] = output_shape[module_id]
+        info['depth'] = depth
+        para = sum(p.numel() for p in module.parameters())
+        info['#param'] = f"{para:,}" if para > 0 else "--"
+        info_list.append(info)
+        if not isinstance(module, ScriptModule):
+            hooks[module_id] = module.register_forward_hook(
+                register_hook(info))
+        if (leaf_module is not None and isinstance(module, leaf_module)):
+            continue
+        module_items = reversed(module._modules.items())
+        stack += [(f"({name}){get_name(mod)}", mod, depth + 1)
+                  for name, mod in module_items if mod is not None]
+
+    training = model.training
+    model.eval()
+    # make a forward pass
+    with torch.no_grad():
+        model(*inputs, **kwargs)
+    model.train(training)
+
+    # remove hooks
+    for h in hooks.values():
+        h.remove()
+
+    info_list = postprocess(info_list)
+    body = make_table(info_list, max_depth=max_depth)
+    return body
+
+
+def get_shape(input: Union[Any, Tuple[Any]]) -> str:
+    if not isinstance(input, tuple):
+        input = (input, )
+    out = []
+    for x in input:
+        if isinstance(x, SparseTensor):
+            out.append(str(list(x.sparse_sizes())))
+        elif hasattr(x, 'size'):
+            out.append(str(list(x.size())))
+    out = ', '.join(out)
+    return out
+
+
+def get_name(module: nn.Module) -> str:
+    return str(module.__class__).split(".")[-1].split("'")[0]
```

Comment:
You can use:
`return module.__class__.__name__`

Reply:
Thanks, much cleaner!